cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/leveldb/LevelDBMetricsTable.java

```diff
@@ -51,8 +54,10 @@
 
   private final LevelDBTableCore core;
 
-  public LevelDBMetricsTable(String tableName, LevelDBTableService service) throws IOException {
-    this.core = new LevelDBTableCore(tableName, service);
+  public LevelDBMetricsTable(DatasetContext datasetContext, String tableName,
+                             LevelDBTableService service, CConfiguration cConf) throws IOException {
+    this.core = new LevelDBTableCore(PrefixedNamespaces.namespace(cConf, datasetContext.getNamespaceId(), tableName),
```

Comment:
We need to construct a name unique across all namespaces for passing to ``BufferingTable`` in the super() call. So keeping this here as opposed to passing all objects to ``BufferingTable`` to construct.